build/yaml/steps/windows/dotnetrestore.yaml

```diff
@@ -22,4 +25,18 @@ steps:
       --configfile
       $(Build.SourcesDirectory)\NuGet.Config
       -bl:"$(Build.ArtifactStagingDirectory)\logs\${{ parameters.BinLogCategory }}_Windows_AnyCPU_$(Configuration)_Restore.binlog"
-  condition: ${{ parameters.Condition }}
+  condition: and(succeeded(), eq(variables['Platform'], 'Any CPU'))
+
+- task: DotNetCoreCLI@2
```

Comment:
Please conditionally only run this for internal builds. Possibly using IsMicroBuildInternal would be an okay differentiation for this.

Reply:
And then conditionally DON'T run the prior step. These steps should be either-or, but not both (although running both probably doesn't cause any issues when running internally).

Reply:
The condition will naturally do an either-or due to the platform being ARM64 or AnyCPU (which we do on separate agents). I'll add an additional IsMicroBuildInternal check here